### ISSUE TITLE
Make heap size flags isolate specific

### ISSUE THREAD

**Issue by zanderso** [dart-lang/sdk#36145]
This would be so that different components running in the same process can specify heap limits independently of each other, either to implement some policy, or for debugging purposes.

@rmacnak-google @a-siva

**Comment by a-siva** [comment]
Do you want this to be something that the embedder specific C++ code should be able to set when creating an isolate or should the spawn API at the Dart level changed to allow for these sizes to be passed as parameters (maybe optional parameters)?

The later would mean exposing internal GC implementation details (presence of new gen and old gen) at the API level which I do not like.

**Comment by zanderso** [comment]
Yeah, I think adding this to the C++ embedding API would be sufficient. Then the flutter_runner and dart_runner can consult some bits in the Fuchsia package to determine the heap limits.
